presto-common/src/main/java/com/facebook/presto/common/block/BlockUtil.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.block;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import javax.annotation.Nullable;
+
+import java.util.Arrays;
+
+import static io.airlift.slice.SizeOf.SIZE_OF_LONG;
+import static java.lang.Math.ceil;
+import static java.lang.Math.max;
+import static java.lang.Math.toIntExact;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public final class BlockUtil
+{
+    private static final double BLOCK_RESET_SKEW = 1.25;
+
+    private static final int DEFAULT_CAPACITY = 64;
+    // See java.util.ArrayList for an explanation
+    static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
+
+    private BlockUtil()
+    {
+    }
+
+    static void checkArrayRange(int[] array, int offset, int length)
+    {
+        requireNonNull(array, "array is null");
+        if (offset < 0 || length < 0 || offset + length > array.length) {
+            throw new IndexOutOfBoundsException(format("Invalid offset %s and length %s in array with %s elements", offset, length, array.length));
+        }
+    }
+
+    static void checkArrayRange(boolean[] array, int offset, int length)
+    {
+        requireNonNull(array, "array is null");
+        if (offset < 0 || length < 0 || offset + length > array.length) {
+            throw new IndexOutOfBoundsException(format("Invalid offset %s and length %s in array with %s elements", offset, length, array.length));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: length (declared at line 50), offset (declared at line 50), array.length. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 54. Consider formatting this data with &#39;%d&#39; instead.